.github/workflows/codeql.yml

```diff
@@ -122,7 +123,44 @@ jobs:
         if: ${{ matrix.goos == 'linux' }}
         run: make bin/vsockexec bin/init
 
+      # only upload results if the analysis fails
+      # otherwise, save the output and use `advanced-security/filter-sarif` to filter paths
       - name: CodeQL Analyze
         uses: github/codeql-action/analyze@v3
         with:
           category: "/language:${{matrix.language}}"
+          output: sarif-results
+          upload: failure-only
+
+      - name: Filter Go SARIF Results
+        uses: advanced-security/filter-sarif@v1
+        with:
+          patterns: |
+            +**/*.go
+            -**/*_test.go
+            -test/**/*.go
+            -vendor/**/*.go
+          input: sarif-results/go.sarif
+          output: sarif-results/go.sarif
+
+      - name: Filter C/C++ SARIF Results
```

Comment:
Why do we need to filter out C/C++ results?

Reply:
We build init and vsockexec when goos=linux (which isn't the most sensical, but 🤷), and want to filter out results from vendored code (or at least keep codeql from warning that those files are ignored)